project/Build.scala

```diff
@@ -56,6 +56,16 @@ object Build extends sbt.Build {
   val newScalaBinaryVersionsInThisRelease: Set[String] =
     Set()
 
+  val javaVersion: Int = {
+    val v = System.getProperty("java.version")
+    v.substring(0, 3) match {
+      case "1.8" => 8
+      case "1.7" => 7
+      case "1.6" => 6
+      case _     => sys.error(s"Unknown JDK version $v")
+    }
```

Comment:
I think this val should be an `Option[Int]`, and we should do something meaningful if we have an unknown JDK, rather than blowing up (possibly printing a warning). What if someone wants to try out Scala.js on beta builds of JDK 9, for example? We recently had a similar problem with the community build (#1965), albeit with the Scala version.